plugin.video.orftvthek/resources/lib/Addon.py

```diff
@@ -199,12 +199,41 @@ def run():
         scraper.getEpisode(link, playlist)
         listCallback(False, pluginhandle)
     elif mode == 'liveStreamRestart':
-        scraper.liveStreamRestart(link)
+        try:
+            import inputstreamhelper
+            is_helper = inputstreamhelper.Helper(input_stream_protocol, drm=input_stream_drm_version)
+            if is_helper.check_inputstream():
+                link = unqoute_url(link)
+                debugLog("Restart Source Link: %s" % link)
+                headers = "User-Agent=%s" % Settings.userAgent()
+                if params.get('lic_url'):
+                    lic_url = unqoute_url(params.get('lic_url'))
+                    debugLog("Playing DRM protected Restart Stream")
+                    debugLog("Restart License URL: %s" % lic_url)
+                    streaming_url, play_item = scraper.liveStreamRestart(link, 'dash')
+                    play_item.setContentLookup(False)
+                    play_item.setMimeType(input_stream_mime)
+                    play_item.setProperty('inputstream.adaptive.stream_headers', headers)
+                    play_item.setProperty('inputstream', is_helper.inputstream_addon)
+                    play_item.setProperty('inputstream.adaptive.manifest_type', input_stream_protocol)
+                    play_item.setProperty('inputstream.adaptive.license_type', input_stream_drm_version)
+                    play_item.setProperty('inputstream.adaptive.license_key', lic_url + '||R{SSM}|')
+                else:
+                    streaming_url, play_item = scraper.liveStreamRestart(link, 'hls')
+                    debugLog("Playing Non-DRM protected Restart Stream")
+                    play_item.setProperty('inputstreamaddon', 'inputstream.adaptive')
+                    play_item.setProperty('inputstream.adaptive.stream_headers', headers)
+                    play_item.setProperty('inputstream.adaptive.manifest_type', 'hls')
+                debugLog("Restart Stream Url: %s" % streaming_url)
+                xbmc.Player().play(streaming_url, play_item)
```

Comment:
why player.play() and not xbmcplugin.setResolvedUrl?

Reply:
i have not been able to get the "Restart" video to work with setResolvedUrl. the video is called from a contextmenu. when i try setResolvedUrl i only get a bunch of "Attempt to use invalid handle -1" but the video doesnt play